Add verifiedCount props to appUser

B77Mills opened the pull request:
Add a new user field for firstVerifiedAt as well as setting a isFirstTimeVerifying flag based on the presence of firstVerifiedAt.

Related PRs: 
 - https://github.com/parameter1/base-cms/pull/733
 - https://github.com/parameter1/randall-reilly-websites/pull/556

B77Mills committed:
Add firstVerifiedAt/isFirstTimeVerifying props to appUser

Add a new user field for firstVerifiedAt as well as setting a isFirstTimeVerifying flag based on the presence of firstVerifiedAt.

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add new event trigger & listener for the first time conversion of a user from non-verified to verified on loginLinkSent/Authenticate.](https://trello.com/c/bS1VKhAo/224-add-new-event-trigger-listener-for-the-first-time-conversion-of-a-user-from-non-verified-to-verified-on-loginlinksent-authentica)

solocommand commented on services/application/src/actions/user/impersonate.js:
```diff
@@ -38,14 +38,23 @@ module.exports = async ({
   if (verify) {
     user.set({
       verified: true,
+      ...(isFirstTimeVerifying && { firstVerifiedAt: new Date() }),
+      isFirstTimeVerifying: false,
```
```suggestion
      verifiedCount: user.verifiedCount + 1,
```

B77Mills committed:
Convert firstVerifiedAt to verifiedCount

Instead of usin firstVerifiedAt & isFirstTimeVerifiying to determine when/if a user is “NEW” use a counter with the default value of 0 and the only time it increments is on login/impersonate.

Meaning the same conditional check can be accounted for if & only if verifiedCount is 1.

B77Mills committed:
Set correct mongo schema type Number

B77Mills committed:
Add MX to prioritize array
